rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1341,70 +1220,99 @@ public BooleanWrapper createNodeSource(String nodeSourceName, String infrastruct
-        String nodeSourceName = nodeSourceData.getName();
+    public void createNotActiveNodeSource(NodeSourceData persistedNodeSource) {
+        InfrastructureManager infrastructureManager = InfrastructureManagerFactory.create(persistedNodeSource);
 
-        //checking that nsname doesn't contain invalid characters and doesn't exist yet
-        checkNodeSourceName(nodeSourceName);
+        NodeSourcePolicy notActivePolicy = NodeSourcePolicyFactory.createNotActive(persistedNodeSource.getPolicyType());
 
-        logger.info("Creating a node source : " + nodeSourceName);
+        NodeSource nodeSource = new NodeSource(this.getUrl(),
+                                               persistedNodeSource.getName(),
+                                               infrastructureManager,
+                                               notActivePolicy,
+                                               (RMCore) PAActiveObject.getStubOnThis(),
+                                               this.monitoring,
+                                               persistedNodeSource.toNodeSourceDescriptor());
 
-        boolean existPersistedNodes = existPersistedNodes(nodeSourceName, nodesRecoverable);
+        String nodeSourceName = nodeSource.getName();
 
-        InfrastructureManager im;
+        undeployedNodeSources.put(nodeSourceName, nodeSource);
 
-        // we need to reload the infrastructure variables saved in database if
-        // we recover the nodes
-        if (existPersistedNodes) {
-            im = InfrastructureManagerFactory.recover(nodeSourceData);
-            if (!im.getDeployingAndLostNodes().isEmpty()) {
-                dbManager.removeAllNodesFromNodeSource(nodeSourceName);
-                im = InfrastructureManagerFactory.create(nodeSourceData);
-                existPersistedNodes = false;
-            }
-        } else {
-            im = InfrastructureManagerFactory.create(nodeSourceData);
-        }
+        this.monitoring.nodeSourceEvent(new RMNodeSourceEvent(RMEventType.NODESOURCE_DEFINED,
+                                                              persistedNodeSource.getProvider().getName(),
+                                                              nodeSourceName,
+                                                              nodeSource.getDescription(),
+                                                              persistedNodeSource.getProvider().getName(),
+                                                              nodeSource.getDescriptor().getStatus().toString()));
 
-        NodeSourcePolicy policy = NodeSourcePolicyFactory.create(nodeSourceData.getPolicyType(),
-                                                                 nodeSourceData.getInfrastructureType(),
-                                                                 nodeSourceData.getPolicyParameters());
+        logger.info("Node source " + nodeSourceName + " has been successfully defined by " +
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Node source " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)